### PR TITLE
migrate phone_log plugin to android v2 embedding.

### PR DESCRIPTION
**Pull request by lidongze91**

### CODE REVIEW

File: packages/phone_log/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## 0.0.7
```
Comment (mehmetf) [inline packages/phone_log/CHANGELOG.md:1]: When you change this; you also need to update pubspec.yaml.

**Comment by mehmetf** [comment]
@zhenqiu1101 PTAL

File: packages/phone_log/CHANGELOG.md (hunk 1 above)
Comment (lidongze91) [reply packages/phone_log/CHANGELOG.md:1]: Done

File: packages/phone_log/android/src/main/java/com/jiajiabingcheng/phonelog/PhoneLogPlugin.java [hunk 2]
```diff
@@ -21,20 +28,71 @@
...
+    context = null;
+  }
+
+  private void attachToActivity(ActivityPluginBinding activityPluginBinding) {
+    this.activity = activityPluginBinding.getActivity();
+    activityPluginBinding.addRequestPermissionsResultListener(this);
+  }
+
+  private void detachToActivity() {
+    this.activity = null;
```
Comment (zhenqiu1101) [inline packages/phone_log/android/src/main/java/com/jiajiabingcheng/phonelog/PhoneLogPlugin.java:77]: Keep the activityPluginBinding and removeRequestPermissionsResultListener when detaching.

Comment (mehmetf) [reply packages/phone_log/android/src/main/java/com/jiajiabingcheng/phonelog/PhoneLogPlugin.java:77]: Ben? :)

**Review by mehmetf — APPROVED** [review]

**Pull request merged**